go/cmd/server/logging/logging_test.go

```diff
@@ -0,0 +1,223 @@
+package logging
+
+import (
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/apache/arrow/go/v8/arrow"
+	"github.com/apache/arrow/go/v8/arrow/array"
+	"github.com/feast-dev/feast/go/internal/feast/model"
+	"github.com/feast-dev/feast/go/internal/test"
+	"github.com/feast-dev/feast/go/protos/feast/serving"
+	"github.com/feast-dev/feast/go/protos/feast/types"
+	"github.com/stretchr/testify/assert"
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+func TestLoggingChannelTimeout(t *testing.T) {
+	// Pregenerated using `feast init`.
+	loggingService, err := NewLoggingService(nil, 1, "", false)
+	assert.Nil(t, err)
+	assert.Empty(t, loggingService.memoryBuffer.logs)
+	ts := timestamppb.New(time.Now())
+	newLog := Log{
+		FeatureStatuses: []serving.FieldStatus{serving.FieldStatus_PRESENT},
+		EventTimestamps: []*timestamppb.Timestamp{ts, ts},
+	}
+	loggingService.EmitLog(&newLog)
+	// Wait for memory buffer flush
+	time.Sleep(20 * time.Millisecond)
+	newTs := timestamppb.New(time.Now())
+
+	newLog2 := Log{
+		FeatureStatuses: []serving.FieldStatus{serving.FieldStatus_PRESENT},
+		EventTimestamps: []*timestamppb.Timestamp{newTs, newTs},
+	}
+	err = loggingService.EmitLog(&newLog2)
+	// The channel times out and doesn't hang.
+	time.Sleep(20 * time.Millisecond)
+	assert.NotNil(t, err)
+}
+
+func TestSchemaTypeRetrieval(t *testing.T) {
+	featureService, entities, featureViews, odfvs := InitializeFeatureRepoVariablesForTest()
+	schema, err := GetSchemaFromFeatureService(featureService, entities, featureViews, odfvs)
+	assert.Nil(t, err)
+	assert.Contains(t, schema.EntityTypes, "driver_id")
+	assert.True(t, reflect.DeepEqual(schema.EntityTypes["driver_id"], types.ValueType_INT64))
+
+	features := []string{"int64", "float32", "int32", "double"}
+	types := []types.ValueType_Enum{*types.ValueType_INT64.Enum(), *types.ValueType_FLOAT.Enum(), *types.ValueType_INT32.Enum(), *types.ValueType_DOUBLE.Enum()}
+	for idx, featureName := range features {
+		assert.Contains(t, schema.FeaturesTypes, featureName)
+		assert.Equal(t, schema.FeaturesTypes[featureName], types[idx])
+	}
+}
+
+func TestSerializeToArrowTable(t *testing.T) {
+	table, err := GenerateLogsAndConvertToArrowTable()
+	assert.Nil(t, err)
+	defer table.Release()
+	tr := array.NewTableReader(table, -1)
+	expected_schema := map[string]arrow.DataType{
+		"driver_id": arrow.PrimitiveTypes.Int64,
+		"int32":     arrow.PrimitiveTypes.Int32,
+		"double":    arrow.PrimitiveTypes.Float64,
+		"int64":     arrow.PrimitiveTypes.Int64,
+		"float32":   arrow.PrimitiveTypes.Float32,
+	}
+
+	expected_columns := map[string]*types.RepeatedValue{
```

Comment:
please use arrow to proto conversion instead of hardcoding this

Reply:
This is the expected schema so I don't think it makes sense to use proto conversion. I refactored so that it exists elsewhere though.